### ISSUE TITLE
Suggestion: a function to check if a NetCDF is a well-formed grid object

### ISSUE THREAD

**Issue by matteodefelice** [SantanderMetGroup/loadeR#24]
Hi, this is a suggestion linked in some way with the issue #22 
I have building a library which uses `loadeR` but I want to be able to deal with also with NetCDF that are not well-formed (files without a grid object). So, the first step would be the use of `loadGridData` but I don't know how to recognize in an efficient way how to check if a file is well-formed. Yes, I could use `dataInventory` catching a possible error like `Error: No variables found` but it would be better to have this detection as a function in `loadeR`, something like `checkNetCDF` or something similar. 
What do you think?

**Comment by sixtohg** [comment]
Hi,
in fact, the function you proposed would be (more or less) these lines:
gds <- openDataset(dataset)
      grid <- gds$findGridByShortName(shortName)
      if (is.null(grid)) {
            stop("Variable requested not found\nCheck 'dataInventory' output and/or dictionary 'identifier'.", call. = FALSE)
}
modifying the error message accordingly to the absence of grids objects instead the variable.
loadGridData is based on the grid object but, although in our opinion is not adequate to loadeR, it is possible, modifying the Java methods applied, to adapt the code of this function to work with the object Variable. This can be done in your library to be used in the case that not grids are defined in the corresponding dataset.
Cheers

**Comment by matteodefelice** [comment]
Exactly, also the function can be extended considering the cases that today lead to an error in `dataInventory`...

**Comment by sixtohg** [comment]
Which function? loadGridData? Now, it is very difficult to extend this function cause it is based in attributes and sub-objects from the grid-object and you have to re-implement all this subroutines to work without this elements. It would be easier to implement other set of functions to read other objects than to adapt or extend loadGridData. Note that the dependence of the grid-object is inherit to the other functions and subroutines of the package and, then, you should to make an structural change.

**Comment by matteodefelice** [comment]
No, I intended a function containing the code lines you have mentioned before. Something like `checkGridData(filename)` returning a boolean.

**Comment by sixtohg** [comment]
ok, sorry I misunderstood. Yes, this function could be easily extended to include other cases.

**Comment by cofinoa** [comment]
@sixtohg and @matteodefelice, please just one more consideration to be added. 

The `gds <- openDataset(dataset)` it's quite expensive because the amount of calls been made to discover and build the collection of well-formed _variables/grids_. 

Therefore the less `openDataset` calls the better. 

The netcdf-java package implements a cache strategy for _small_ size variables (and attributes) but I'm not sure if this cache is smart enough to hold all the Grid discoveries roundtrips.

**Comment by matteodefelice** [comment]
Maybe we can modify `dataInventory` to return NaN in case of malformed NetCDF, what do you think?
